htlcswitch/queue.go

```diff
@@ -152,9 +152,18 @@ func (p *packetQueue) packetCoordinator() {
 			case p.outgoingPkts <- nextPkt:
 				// Only decrease the queueLen and totalHtlcAmt once the packet has been
 				// sent out
-				debug_print("going to remove nextPkt from the queue")
+				// FIXME: do we need these to be atomic? Since the queue is per channel
+				// link, I think only one thread can be updating this at a time?
 				atomic.AddInt32(&p.queueLen, -1)
 				atomic.AddInt64(&p.totalHtlcAmt, int64(-nextPkt.amount))
+				// update the minHtlc value
+				for i := 0; i < int(p.Length()); i++ {
```

Comment:
@vibhaa yeah this wasn't entirely clear - might depend on what dequeuing strategy we want to use / and whether we want to expose the current balance to the queue manager or not. Currently, the queue manager does know about the current balance, and every 10 ms, a goroutine in link.go decides whether to signal to the queue or not. One option is to do this every time there is a change in balance on the channel - but this may have a couple of problems: 
a) it could lead to many spurious wake ups
b) if a change in balance makes it possible to send many payments from the overflowQueue, then this might not function correctly. 

Instead, we just keep track of the minimum htlc amount, so it avoids spurious signals to the queue, while also always signaling to the queue if it may be possible to send at send at least some payment. In case the queue decides to send a payment for which there aren't enough funds - this still does not lead to incorrect behavior as the new payment will simply be added to the back of the queue again by the link. This won't create head of line blocking because it is added to the back of the queue - but it does create an annoying delay (because you can only dequeue one payment per 10 ms, or whatever frequency we choose)